components/recurring-contributions/UpdatePaymentMethodPopUp.js

```diff
@@ -0,0 +1,280 @@
+      return <MoneyCheck width="26px" height="18px" />;
+    }
+  };
+
+  const getPaymentMethodMetadata = pm => {
+    if (pm.type === 'creditcard') {
+      const expiryDate = paymentMethodExpiration(pm);
+      return (
+        <FormattedMessage
+          id="ContributePayment.expiresOn"
+          defaultMessage="Expires on {expiryDate}"
+          values={{ expiryDate }}
+        />
+      );
+    } else if (pm.type === 'virtualcard') {
+      if (pm.balance < minBalance) {
+        return (
+          <FormattedMessage
+            id="ContributePayment.unusableBalance"
+            defaultMessage="{balance} left, balance less than {minBalance} cannot be used."
+            values={{
+              balance: formatCurrency(pm.balance, pm.currency),
+              minBalance: formatCurrency(minBalance, pm.currency),
+            }}
+          />
+        );
+      } else if (pm.expiryDate) {
+        return (
+          <FormattedMessage
+            id="RecurringContributions.balanceAndExpiry"
+            defaultMessage="{balance} left, expires {expiryDate}"
+            values={{
+              expiryDate: <FormattedDate value={pm.expiryDate} month="numeric" year="numeric" />,
+              balance: formatCurrency(pm.balance, pm.currency),
+            }}
+          />
+        );
+      } else {
+        return (
+          <FormattedMessage
+            id="ContributePayment.balanceLeft"
+            defaultMessage="{balance} left"
+            values={{ balance: formatCurrency(pm.balance, pm.currency) }}
+          />
+        );
+      }
+    } else if (['prepaid', 'collective'].includes(pm.type)) {
+      return (
+        <FormattedMessage
+          id="ContributePayment.balanceLeft"
+          defaultMessage="{balance} left"
+          values={{ balance: formatCurrency(pm.balance, pm.currency) }}
+        />
+      );
+    }
+  };
+
+  const generatePaymentsOptions = () => {
+    // Add collective payment methods
+    const paymentMethodsOptions = get(data, 'Collective.paymentMethods', [])
+      .filter(pm => !(pm.service === 'paypal' && pm.type === 'adaptive'))
+      .map(pm => ({
+        key: `pm-${pm.id}`,
+        title: getPaymentMethodName(pm),
+        subtitle: getPaymentMethodMetadata(pm),
+        icon: getPaymentMethodIcon(pm),
+        paymentMethod: pm,
+        disabled: pm.balance < minBalance,
+        id: pm.id,
+      }));
+
+    return uniqBy(paymentMethodsOptions, 'key');
+  };
+
+  const paymentOptions = generatePaymentsOptions();
+
+  return (
+    <Fragment>
+      <Flex width={1} alignItems="center" justifyContent="center" minHeight={45}>
+        <P my={2} fontSize="Caption" textTransform="uppercase" color="black.700">
+          {intl.formatMessage(messages.updatePaymentMethod)}
+        </P>
+        <Flex flexGrow={1} alignItems="center">
+          <StyledHr width="100%" mx={2} />
+        </Flex>
+        <PlusCircle size={20} />
+      </Flex>
+      <StyledRadioList
```

Comment:
I was unsure whether this needed to be wrapped in a `<StyledFieldInput>`. I saw radio lists both with and without in the codebase, and ended up basing mine on the one in the Contribution flow, which doesn't have it.

Reply:
As per its description, `StyledInputField` is mainly used to display labels and errors. There's no imperative to use it, but it's helpful when you need to do that.